### PR TITLE
File

### PR DESCRIPTION
**Pull request by mcardosos**
Works with changes on https://github.com/amarzavery/sway/pull/16

### CODE REVIEW

**Comment by mcardosos** [comment]
Fixes https://github.com/Azure/oav/issues/167 and https://github.com/Azure/oav/issues/136

File: lib/validators/specValidator.js [hunk 1]
```diff
@@ -643,6 +643,21 @@ class SpecValidator {
...
+        if (operation.consumes) {
+          if (!options.headers) options.headers = {};
+
+          function isOctetStream(consumes) {
+            return consumes.some((contentType) => {
+              return contentType === 'application/octet-stream';
+            });
+          }
+
+          if (parameter.schema.format === 'file' && isOctetStream(operation.consumes)) {
```
Comment (amarzavery) [inline lib/validators/specValidator.js]: what about `parameter.schema.type === 'file'` ?

Comment (amarzavery) [reply lib/validators/specValidator.js]: what about **only** `parameter.schema.type === 'object'` **No "format": "file"` present** ?

File: test/modelValidation/swaggers/specification/contenttype/datalake.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          "Uploads stream": {
+            "$ref": "./examples/upload.json"
+          }
+        },
+        "parameters": [
+          {
+            "name": "streamContents",
+            "in": "body",
+            "schema": {
+              "type": "object",
```
Comment (amarzavery) [inline test/modelValidation/swaggers/specification/contenttype/datalake.json]: I don't see a test case for `"type": "file"`

**Comment by amarzavery** [comment]
## Test Matrix
| where | type| format  |
|---|---|---|
| request/response  | object | file  |
| request/response  |  file |   |

**Comment by amarzavery** [comment]
@mcardosos - The very famous Mariana, whenever you find time, please complete this as some teams are waiting on this.

File: lib/validators/specResolver.js [hunk 3]
```diff
@@ -536,7 +536,9 @@ class SpecResolver {
           for (let statusCode in operation.responses) {
             let response = operation.responses[statusCode];
             if (response.schema) {
-              response.schema = utils.relaxModelLikeEntities(response.schema);
+              if (operation.produces != 'application/octet-stream') {
```
Comment (amarzavery) [inline lib/validators/specResolver.js]:
```javascript
let produces = operation.produces || specInJson.produces;
if (produces && !produces.some((item) => { return item.toLowerCase() === 'application/octet-stream'; })) {
  response.schema = utils.relaxModelLikeEntities(response.schema); 
}
```

**Pull request merged**